src/server/service/user.ts

```diff
@@ -30,7 +30,51 @@ export default class UserService implements IUserService {
     return hash;
   }
 
-  async create(dto: CreateUserDTO): Promise<User> {
+  private checkRoles(role: string): void {
+    const ROLES = [
+      Role.Admin as string,
+      Role.Manager as string,
+      Role.User as string,
+    ];
+
+    if (!ROLES.includes(role)) {
+      throw new InvalidUserRole();
+    }
+  }
+
+  private async checkPermission(email: string, role: Role) {
+    const user = await UserModel.findOne({ email });
+
+    if (!user) {
+      throw new Error(`User with email "${email}" does not exists`);
+    }
+
+    if (user.role === Role.Admin) {
+      if (role !== Role.Manager) {
+        throw new Error(
+          'Admin user does not have permission for creating this user type',
+        );
```

Comment:
We have the same issue here as we do there: https://github.com/EstebanBorai/thruway-parking/pull/7#discussion_r632893209

We are relying on primitives and generic `Error` again, and thats harder to handle and to maintain.
I suggest implementing a dedicated error.